sigal/gallery.py

```diff
@@ -165,6 +167,23 @@ def exif(self):
         return (get_exif_tags(self.raw_exif, datetime_format=datetime_format)
                 if self.raw_exif and self.ext in ('.jpg', '.jpeg') else None)
 
+    def _get_metadata(self):
+        super(Image, self)._get_metadata()
+        # If a title or description hasn't been obtained by other means, look
+        #  for the information in IPTC fields as catalogued in:
+        # https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata
+        if not self.title or not self.description:
+            iptc_data = get_iptc_data(self.src_path)
+            if iptc_data and not self.title:
+                # 2:05 is the IPTC title property
+                if (2, 5) in iptc_data:
+                    self.title = iptc_data[(2, 5)].decode('utf-8')
+
+            if iptc_data and not self.description:
+                # 2:120 is the IPTC description property
+                if (2, 120) in iptc_data:
+                    self.description = iptc_data[(2, 120)].decode('utf-8')
```

Comment:
I think you could put the details of how to get title and description in `get_iptc_data`, it would make the function easier to test and more self-contained, and avoid to do this directly in `Image` which does not have to know how IPTC work.

Reply:
Good idea.